internal/util/prometheus.go

```diff
@@ -0,0 +1,55 @@
+package util
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+type ControllerFunctionalPrometheusMetrics struct {
+	// number of post /config to proxy successfully
+	ConfigPassCounter prometheus.Counter
+	// number of post /config to proxy failed
+	ConfigFailureCounter prometheus.Counter
+	// number of ingress analysis failure
+	ParseFailureCounter prometheus.Counter
+
+	// duration of last successful confiuration sync
+	ConfigureDurationHistogram prometheus.Histogram
```

Comment:
For consistency with `proxy_configuration_count`, please call this metric `proxy_configuration_duration_milliseconds` (if you use milliseconds, or change the unit accordingly) and use the exact same labels.

Reply:
Good idea ! 👍